teps/0111-replace-volumes-with-workspaces.md

```diff
@@ -0,0 +1,350 @@
+and the TaskRun can supply a secret or configMap in the [workspace binding](https://tekton.dev/docs/pipelines/workspaces/#specifying-volumesources-in-workspaces).
+
+These secrets can also be isolated to only the Steps that need them using
+[isolated workspaces](https://tekton.dev/docs/pipelines/workspaces/#isolating-workspaces-to-specific-steps-or-sidecars)
+(note: currently in alpha).
+
+#### Caching data across multiple Steps
+
+An example use case for caching data across multiple steps is image build Tasks that cache image layers.
+This use case can be met using a workspace with an [emptyDir binding](https://tekton.dev/docs/pipelines/workspaces/#emptydir).
+
+#### Docker in Docker (most CI/CD use cases)
+
+Docker-in-Docker Tasks typically run a Docker daemon in a Sidecar.
+[Workspaces may be mounted into Sidecars](https://tekton.dev/docs/pipelines/workspaces/#sharing-workspaces-with-sidecars),
+a feature that allows data (such as TLS certificates) to be shared between the Docker daemon and the build Step.
+(Note: workspaces in sidecars are currently in alpha.)
+
+Most CI/CD use cases, such as those that involve building Docker images, can be supported
+by running a Docker daemon in a sidecar (the approach used by the catalog 
+[docker-build Task](https://github.com/tektoncd/catalog/blob/main/task/docker-build/0.1/docker-build.yaml)).
+The build container can connect to the sidecar via TLS, or by mounting an emptyDir volume
+to access the sidecar's docker socket.
+This approach requires using privileged containers, but does not require mounting any files
+from the host node.
+No image build Tasks in the Catalog mount files from host nodes.
+
+### Volume use cases not currently supported by Workspaces
+
+#### Docker in Docker (advanced use cases)
+
+Some users would like to be able to access the host node's image cache for performing Docker builds.
+This requires mounting the host's
+[docker socket](https://docs.docker.com/engine/reference/commandline/dockerd/#daemon-socket-option).
+The most natural option to do so is a [hostPath volume](https://kubernetes.io/docs/concepts/storage/volumes/#hostpath).
+Workspaces do not support hostPath volumes as bindings, so this feature would not be supported
+by Workspaces. However, this approach is not recommended for
+[security reasons](https://blog.quarkslab.com/kubernetes-and-hostpath-a-love-hate-relationship.html).
+
+The only Catalog Task using hostPath volumes is [`kind`](https://hub.tekton.dev/tekton/task/kind).
+For more information on why `kind` on Kubernetes requires hostPath volumes, see
+https://github.com/kubernetes-sigs/kind/issues/303#issuecomment-521384993 and
+[Running KIND Inside A Kubernetes Cluster For Continuous Integration](https://d2iq.com/blog/running-kind-inside-a-kubernetes-cluster-for-continuous-integration)
+(TL;DR: because `kind` needs to create nested Docker containers).
```

Comment:
What would be the alternative here if we removed workspaces?
I agree that using volumes directly instead of workspaces is not a best practice, but unless we have an alternative here I wonder if we could address this by documentation, i.e. make sure that volumes are not used in any example / catalog task unless were strictly required. We could add a special page / section to document this feature as something we do not recommend to use unless in specific documented cases and specify security risks.

Reply:
I've added this as an option to "alternatives". Right now, I don't think we would have an alternative for this using workspaces.